packages/integrations/src/media-organizer/lidarr/lidarr-integration.ts

```diff
@@ -0,0 +1,140 @@
+import { logger } from "@homarr/log";
+import { z } from "@homarr/validation";
+
+import type { CalendarEvent } from "../../calendar-types";
+import { MediaOrganizerIntegration } from "../media-organizer-integration";
+
+export class LidarrIntegration extends MediaOrganizerIntegration {
+  public async testConnectionAsync(): Promise<void> {
+    await super.handleTestConnectionResponseAsync({
+      queryFunctionAsync: async () => {
+        return await fetch(`${this.integration.url}/api`, {
+          headers: { "X-Api-Key": super.getSecretValue("apiKey") },
+        });
+      },
+    });
+  }
+
+  /**
+   * Gets the events in the Lidarr calendar between two dates.
+   * @param start The start date
+   * @param end The end date
+   * @param includeUnmonitored When true results will include unmonitored items of the Tadarr library.
+   */
+  async getCalendarEventsAsync(start: Date, end: Date, includeUnmonitored = true): Promise<CalendarEvent[]> {
+    const url = new URL(this.integration.url);
+    url.pathname = "/api/v1/calendar";
+    url.searchParams.append("start", start.toISOString());
+    url.searchParams.append("end", end.toISOString());
+    url.searchParams.append("unmonitored", includeUnmonitored ? "true" : "false");
+    const response = await fetch(url, {
+      headers: {
+        "X-Api-Key": super.getSecretValue("apiKey"),
+      },
+    });
+    const lidarrCalendarEvents = await z.array(lidarrCalendarEventSchema).parseAsync(await response.json());
+
+    return lidarrCalendarEvents.map((lidarrCalendarEvent): CalendarEvent => {
+      return {
+        name: lidarrCalendarEvent.title,
+        subName: lidarrCalendarEvent.artist.artistName,
+        description: lidarrCalendarEvent.overview,
+        thumbnail: this.chooseBestImageAsURL(lidarrCalendarEvent),
+        date: lidarrCalendarEvent.releaseDate,
+        mediaInformation: {
+          type: "audio",
+        },
+        links: this.getLinksForLidarrCalendarEvent(lidarrCalendarEvent),
+      };
+    });
+  }
+
+  private getLinksForLidarrCalendarEvent = (event: z.infer<typeof lidarrCalendarEventSchema>) => {
+    const links: CalendarEvent["links"] = [];
```

Comment:
Use a map on event.artist.links and switch statement instead? This avoids all the non-null assertions.